### PR TITLE
use netcgo resolver by default on linux

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

Force the use of the `cgo` DNS resolver instead of the pure Golang one. 

This is done by enabling the `netcgo` build flag. This flag has the effect to enable this file at compilation:
https://github.com/golang/go/blob/master/src/net/conf_netcgo.go

This will force usage of `cgo`:
https://github.com/golang/go/blob/2e9f0817f070a3979afc580c740670690acab672/src/net/conf.go#L85

It seems that this behavior can be overwritten at runtime by setting the `GODEBUG` env var like so:
```
GODEBUG=netdns=go
```
https://github.com/golang/go/blob/2e9f0817f070a3979afc580c740670690acab672/src/net/conf.go#L129-L135

However, this might be impractical depending on how the agent is started.

### Motivation

The Golang resolver has some minor difference compared to `glibc` implementation leading to inconsistent behavior. More info can be found [here](https://engineering.grab.com/dns-resolution-in-go-and-cgo).

### CODE REVIEW

**Review by remicalixte — DISMISSED** [review]

**Comment by remicalixte** [comment]
It seems to work great. I didn't check the `glibc` behavior though

File: releasenotes/notes/use-netcgo-dns-68d9fbad3b7c01a9.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+other:
+  - |
+    Use the cgo dns resolver instead of the golang one. The will make the agent
+    use glibc to resolve hostnames and should give more predicatable results.
```
Comment (olivielpeau) [inline releasenotes/notes/use-netcgo-dns-68d9fbad3b7c01a9.yaml]: typo: `predictable`

File: tasks/build_tags.py [hunk 2]
```diff
@@ -37,6 +37,7 @@
     "kubelet",
     "kubeapiserver",
     "cri",
+    "netcgo",
```
Comment (olivielpeau) [inline tasks/build_tags.py]: I think this tag deserves a small comment since it's not a tag we define ourselves :)

**Review by olivielpeau — DISMISSED** [review]
a couple of nits, but other than that looks good!

This would affect logs, let's ask if it's fine by them as well.

Let's also ping trace and process. And container-integration for the DCA

**Review by olivielpeau — APPROVED** [review]

**Pull request merged**